### PR TITLE
Add Interaction Localization support

### PR DESCRIPTION
**Pull request by OurTwinkie**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing
[pull-request]: https://github.com/JDA-Applications/JDA-Utilities/pulls

#### Pull Request Checklist
Please follow the following steps before opening this PR.<br>
PRs that do not complete the checklist will be subject to denial for
missing information.

- [ ] I have checked the [pull request page][pull-request] for upcoming
      or merged features/bug fixes.
- [ ] I have read JDA's [contributing guidelines][contributing].

#### Pull Request Information
Check and fill in the blanks for all that apply:

- [ ] My PR fixes a bug, error, or other issue with the library's codebase.
- [ ] My PR is for the `______` module of the JDA-Utilities library.
- [ ] My PR creates a new module for the JDA-Utilities library: `______`.

#### Description

Describe your PR here.

### CODE REVIEW

File: command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java [hunk 1]
```diff
@@ -72,6 +68,11 @@
  */
 public abstract class SlashCommand extends Command
 {
+
```
Comment (Chew) [inline command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java]: Docs

File: command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java [hunk 2]
```diff
@@ -72,6 +68,11 @@
...
+    protected Map<DiscordLocale, String> nameLocalization = new HashMap<>();
+
```
Comment (Chew) [inline command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java:82]: Docs

File: command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java [hunk 3]
```diff
@@ -505,4 +517,12 @@ else if(cooldownScope.equals(CooldownScope.GUILD) && event.getGuild()==null)
         else
             return front+" "+cooldownScope.errorSpecification+"!";
     }
+
```
Comment (Chew) [inline command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java:555]: Docs

File: command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java [hunk 4]
```diff
@@ -505,4 +517,12 @@ else if(cooldownScope.equals(CooldownScope.GUILD) && event.getGuild()==null)
...
+    public Map<DiscordLocale, String> getNameLocalization() {
+        return nameLocalization;
+    }
+
```
Comment (Chew) [inline command/src/main/java/com/jagrosh/jdautilities/command/SlashCommand.java:563]: Docs

**Review by Chew — CHANGES_REQUESTED** [review]
Great work! I'll need to do some testing then we'll be good to go.

I may want a helper to help make the locales, but we'll see.

**Comment by Chew** [comment]
Do menus also have locales?

**Comment by OurTwinkie** [comment]
> Do menus also have locales?

No, today I will do the same for the menu and write the docs.

**Review by Chew — APPROVED** [review]
Seems good! Gonna do some testing just to be sure.

**Comment by Chew** [comment]
image
Weird, I added the name localization to a parent command, but it didn't take effect for the child. Not sure what's up. Nor did description locale take effect.

**Comment by OurTwinkie** [comment]
> image
> 
> Weird, I added the name localization to a parent command, but it didn't take effect for the child. Not sure what's up. Nor did description locale take effect.

Fixed it, now it works.

**Comment by Chew** [comment]
image
Looks great!

**Pull request merged**